File: lib-base/src/main/java/com/llj/lib/base/IUiHandler.java
package com.llj.lib.base;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.TextView;

import com.llj.lib.base.listeners.OnMyClickListener;
import com.llj.lib.utils.ACollectionUtils;
import com.llj.lib.utils.ATextUtils;
import com.llj.lib.utils.AToastUtils;

import java.util.Collection;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public interface IUiHandler {

    ///////////////////////////////////////////////////////////////////////////
    // ui相关操作
    ///////////////////////////////////////////////////////////////////////////


    default void setOnClickListener(View view, @Nullable View.OnClickListener onClickListener) {
        if (onClickListener == null) {
            view.setOnClickListener(null);
        } else {
            view.setOnClickListener(new OnMyClickListener() {
                @Override
                public void onCanClick(View v) {
                    onClickListener.onClick(v);
                }
            });
        }
    }

    default CharSequence getTextStr(TextView textView) {
        return ATextUtils.getText(textView);
    }


    default void setText(TextView textView, CharSequence destination) {
        ATextUtils.setText(textView, destination);
    }

    default void setText(TextView textView, CharSequence destination, CharSequence defaultStr) {
        ATextUtils.setText(textView, destination, defaultStr);
    }

    default void showToast(String content) {
        AToastUtils.show(content);
    }

    default void showToast(@StringRes int resId) {
        AToastUtils.show(resId);
    }

    default void showLongToast(String content) {
        AToastUtils.showLong(content);
    }

    default void showLongToast(@StringRes int resId) {
        AToastUtils.showLong(resId);
    }

    default boolean isEmpty(CharSequence text) {
        return android.text.TextUtils.isEmpty(text);
    }

    default boolean isEmpty(TextView textView) {
        return android.text.TextUtils.isEmpty(getTextStr(textView));
    }

    default boolean isEmpty(Collection list) {
        return ACollectionUtils.isEmpty(list);
    }

    default void setTextColor(TextView textView, @ColorRes int id) {
        textView.setTextColor(ContextCompat.getColor(textView.getContext(), id));
    }

    default int getCompatColor(Context context, @ColorRes int id) {
        return ContextCompat.getColor(context, id);
    }


    default Drawable getCompatDrawable(Context context, @DrawableRes int id) {
        return ContextCompat.getDrawable(context, id);
    }


}

File: lib-base/src/main/java/com/llj/lib/base/IUiHandler.kt
package com.llj.lib.base

import android.content.Context
import android.graphics.drawable.Drawable
import android.graphics.drawable.GradientDrawable
import android.view.View
import android.widget.TextView
import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import androidx.core.content.ContextCompat
import com.llj.lib.utils.ACollectionUtils
import com.llj.lib.utils.ADisplayUtils
import com.llj.lib.utils.ATextUtils
import com.llj.lib.utils.AToastUtils

/**
 * ArchitectureDemo
 * describe: 相关ui控制
 * @author llj
 * @date 2018/5/24
 */
interface IUiHandler {


  ///////////////////////////////////////////////////////////////////////////
  // ui相关操作
  ///////////////////////////////////////////////////////////////////////////

  fun getTextStr(textView: TextView?): String {
    return ATextUtils.getText(textView)
  }

  fun setText(textView: TextView?, destination: CharSequence?, defaultStr: CharSequence? = null) {
    ATextUtils.setText(textView, destination, defaultStr)
  }

  fun setTextWithVisibility(textView: TextView?, destination: CharSequence?) {
    ATextUtils.setTextWithVisibility(textView, destination)
  }

  fun showToast(content: CharSequence?) {
    AToastUtils.show(content)
  }

  fun showToast(@StringRes resId: Int) {
    AToastUtils.show(resId)
  }

  fun showLongToast(content: CharSequence?) {
    AToastUtils.showLong(content)
  }

  fun showLongToast(@StringRes resId: Int) {
    AToastUtils.showLong(resId)
  }

  fun isEmpty(text: CharSequence?): Boolean {
    return android.text.TextUtils.isEmpty(text)
  }

  fun isEmpty(textView: TextView): Boolean {
    return android.text.TextUtils.isEmpty(getTextStr(textView))
  }

  fun isEmpty(list: Collection<Any?>?): Boolean {
    return ACollectionUtils.isEmpty<Any>(list)
  }

  fun setTextColor(textView: TextView, @ColorRes id: Int) {
    textView.setTextColor(ContextCompat.getColor(textView.context, id))
  }

  fun getCompatColor(context: Context, @ColorRes id: Int): Int {
    return ContextCompat.getColor(context, id)
  }

  fun getCompatDrawable(context: Context, @DrawableRes id: Int): Drawable? {
    return ContextCompat.getDrawable(context, id)
  }


  fun getCustomDrawable(
      context: Context,
      @ColorRes bgColor: Int,
      @ColorRes strokeColor: Int,
      connerLAT: Float = 0F, connerRAT: Float = 0F, connerRAB: Float = 0F, connerLAB: Float = 0F,
      strokeWidth: Int = 0,
      dashWidth: Float = 0F,
      dashGap: Float = 0F,
      shape: Int = GradientDrawable.RECTANGLE,
  ): Drawable {
    val drawable = GradientDrawable()
    drawable.shape = shape
    drawable.setColor(getCompatColor(context, bgColor))
    drawable.cornerRadii = floatArrayOf(connerLAT, connerLAT, connerRAT, connerRAT, connerRAB, connerRAB, connerLAB, connerLAB)
    drawable.setStroke(strokeWidth, getCompatColor(context, strokeColor), dashWidth, dashGap)
    return drawable
  }


  fun dip2px(context: Context, dpValue: Float): Int {
    return ADisplayUtils.dp2px(context, dpValue)
  }

  fun nullToEmpty(destination: CharSequence?): CharSequence {
    return destination ?: ""
  }

  fun getMiddle(value: Float, minValue: Float, maxValue: Float): Float {
    return Math.min(maxValue, Math.max(minValue, value))
  }

  fun getMiddle(value: Double, minValue: Double, maxValue: Double): Double {
    return Math.min(maxValue, Math.max(minValue, value))
  }

  fun getMiddle(value: Int, minValue: Int, maxValue: Int): Int {
    return Math.min(maxValue, Math.max(minValue, value))
  }


  fun isVisible(view: View): Boolean {
    return view.visibility == View.VISIBLE
  }

  fun isInVisible(view: View): Boolean {
    return view.visibility == View.INVISIBLE
  }

  fun isNotVisible(view: View): Boolean {
    return view.visibility == View.INVISIBLE || view.visibility == View.GONE
  }

  fun isGone(view: View): Boolean {
    return view.visibility == View.GONE
  }

}
